### ISSUE TITLE
Animation isn't happening till entire page loads

### ISSUE THREAD

**Issue by mouryareddy** [danilowoz/react-content-loader#257]
The animation should happen as soon as possible. Instead, it is happening after the entire page is loaded. Till the entire page loads, It's static.

**Comment by fandiario** [comment]
Have some problem with this. I'm using next js and also set SSR to false... Wonder where did i go wrong...

![image](https://user-images.githubusercontent.com/66576670/132217274-3c65dbaa-f517-46fe-81a9-caa779997aca.png)

and my skeleton loader

![image](https://user-images.githubusercontent.com/66576670/132217360-360de544-ede4-4f3f-9c13-4f31cc2f09c8.png)

I'm using next 11.0.1 and  react-content-loader 6.0.3

**Comment by kalinkochnev** [comment]
I had the same issue. First make sure that you don't have any blocking function calls, but I think the issue has to do something with the use of svg animations vs css. I had other css animations that were working fine, but this package got hung until the async functions finished.

**Comment by mouryareddy** [comment]
I am using a fetch call , and loading this till I get result. Do you have that CSS property handy so that I could add it.

**Comment by kalinkochnev** [comment]
> I am using a fetch call , and loading this till I get result. Do you have that CSS property handy so that I could add it.

Sorry I probably didn't explain what I meant clearly. I was only saying that css animations appear to work while the page is loading, but the SVG ones do not. I haven't figured the issue out either.

**Comment by RohithBhandaru** [comment]
I'm also facing similar issue. I have a screen which has a graph with large data and a list with corresponding data points to be rendered. I wanted to place content loader while the graph screen gets rendered. However, the content loader animation doesn't even start before the graph screen gets rendered. I'm new to react native.

Am I missing something here? Any help is appreciated.

**Comment by hubsmoke** [comment]
Also having this issue, I think it's related to browsers not prioritizing SVG animations. If this is correct, this project should offer a CSS solution by overlaying 2 SVGs and using an animation on the css mask-position to simulate the loading animation. I may cook this up as a PR if I don't find another solution

**Comment by cwveysey** [comment]
Also having this problem -

**Comment by kostas64** [comment]
Same problem. animation takes tool long before start

**Comment by nenorrell** [comment]
Also having this issue. It's problematic when using this as a fallback for Suspense components, as there is no animation since it doesn't animate until the page loads (at which point, the fallback is no longer needed)

**Comment by 27shutterclicks** [comment]
Same here, svg and lottie animations weren't animating until async data fetches complete, so I couldn't get this component to work - on Next.js with SSR. I switched to _react-loading-skeleton_, thanks to @hubsmoke 's suggestion above and it's working great.

**Comment by shehi** [comment]
Yup... With Next.js 14 and RSC (react server components) this ain't working: page won't finish loading until async processes are done (Suspension) and when that happens, react-content-loader is gone anyway. Makes this component's animations impossible to show.
